impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultDependencyResolverResult.java

```diff
@@ -97,6 +99,24 @@ public class DefaultDependencyResolverResult implements DependencyResolverResult
      */
     private final PathModularizationCache cache;
 
+    /**
+     * Creates an initially empty result with a temporary cache.
+     * Callers should add path elements by calls to {@link #addDependency(Node, Dependency, Predicate, Path)}.
+     *
+     * <p><b>WARNING: this constructor may be removed in a future Maven release.</b>
+     * The reason is because {@code DefaultDependencyResolverResult} needs a cache, which should
+     * preferably by session-wide. But we have not yet clarified how such caches should be managed.</p>
+     *
+     * @param request the corresponding request
+     * @param exceptions the exceptions that occurred while building the dependency graph
+     * @param root the root node of the dependency graph
+     * @param count estimated number of dependencies
+     */
+    public DefaultDependencyResolverResult(
+            DependencyResolverRequest request, List<Exception> exceptions, Node root, int count) {
+        this(request, new PathModularizationCache(), exceptions, root, count);
```

Comment:
yes this gives dedication and option.

Reply:
```suggestion
        this(request, moduleCache(), exceptions, root, count);
```

Reply:
The `moduleCache()` method is defined in another class. We cannot access that method from here.